### PR TITLE
Deprecated default_report_months

### PR DESCRIPTION
**Pull request by divyaac**
Matching ENT PR: https://github.com/hashicorp/vault-enterprise/pull/5929

closes https://github.com/hashicorp/vault-enterprise/pull/5929

### CODE REVIEW

File: website/content/docs/upgrading/index.mdx
Comment (mpalmi) [inline website/content/docs/upgrading/index.mdx:139]: I think we probably want to use a version-based doc, like this: https://github.com/hashicorp/vault/blob/main/website/content/docs/upgrading/upgrade-to-1.17.x.mdx

I don't think there's a pre-existing 1.18 upgrade guide, but if you take a look at the docs-related changes in https://github.com/hashicorp/vault/commit/0637f5e316709c7f2eb0c5e3d6ad7030a8ef9e55#diff-98be8c35d45b7eff158ef4d272e6c3092c4501a1a88db1ee907365316fd6bbd4 it should show you what's needed:

1. Add the new file/update version numbers.
2. Update the docs nav .json file

File: vault/external_tests/activity/activity_test.go
Comment (mpalmi) [inline vault/external_tests/activity/activity_test.go:71]: I'm not 100% certain, but I don't think this should be needed. The activity export should display empty months, but will also include the field that we're after (default_report_months).

**Review by mpalmi — COMMENTED** [review]
Just a couple of things that I think should be addressed before merging.

File: website/content/docs/upgrading/index.mdx
Comment (divyaac) [reply website/content/docs/upgrading/index.mdx:139]: Just added this!

File: vault/external_tests/activity/activity_test.go
Comment (divyaac) [reply vault/external_tests/activity/activity_test.go]: oops this was mistake, I removed this change (should be in another pr instead)

File: website/content/docs/upgrading/index.mdx
Comment (mpalmi) [reply website/content/docs/upgrading/index.mdx:139]: We probably only want it one place, so we might want to remove the change to this file.

File: website/content/docs/upgrading/upgrade-to-1.18.x.mdx
Comment (mpalmi) [inline website/content/docs/upgrading/upgrade-to-1.18.x.mdx:20]:
```suggestion
The `default_report_months` field can no longer be configured via the
[/sys/internal/counters/config](/vault/api-docs/system/internal-counters#update-the-client-count-configuration) endpoint.

The endpoint will now return a warning:

```shell
 <code-quoted example>
```shell
 
```

Comment (mpalmi) [reply website/content/docs/upgrading/upgrade-to-1.18.x.mdx:20]: ^ It might be useful to have a bit more information for any users in case they encounter the error message (and also link to the endpoint for some context.

File: website/content/api-docs/system/internal-counters.mdx
Comment (mpalmi) [inline website/content/api-docs/system/internal-counters.mdx:1100]: @schavis do you have any specific style guidance for deprecated fields?

Comment (schavis) [inline website/content/api-docs/system/internal-counters.mdx:1100]:
```suggestion
- `default_report_months` `(integer: 12)` - **DEPRECATED** The number
   of months to report if no `start_time` is specified in a query.
```

Comment (schavis) [reply website/content/api-docs/system/internal-counters.mdx:1100]: Added a suggestion :)

File: website/content/docs/upgrading/index.mdx
Comment (schavis) [inline website/content/docs/upgrading/index.mdx:139]:
```suggestion
<Warning title="Read-only field">

  You cannot configure the `default_report_months` field. Vault will ignore all
  write requests.
  
</Warning>
```

Suggest making the warning more visible with an aside.

File: website/content/docs/upgrading/upgrade-to-1.18.x.mdx
Comment (schavis) [inline website/content/docs/upgrading/upgrade-to-1.18.x.mdx:20]: Slight tweak to the suggestion made above:

````suggestion
The `default_report_months` field is now read-only.

If you attempt to modify `default_report_months` through the
[/sys/internal/counters/config](/vault/api-docs/system/internal-counters#update-the-client-count-configuration)
endpoint, Vault ignores the update and returns a warning:

<CodeBlockConfig hideClipboard>

```shell-session
<code-quoted example>
```

</CodeBlockConfig>

````

**Review by schavis — CHANGES_REQUESTED** [review]

File: website/content/docs/upgrading/index.mdx
Comment (mpalmi) [inline website/content/docs/upgrading/index.mdx:139]: Looks like we still need to drop this:

```suggestion
```

File: website/content/docs/upgrading/upgrade-to-1.18.x.mdx
Comment (mpalmi) [inline website/content/docs/upgrading/upgrade-to-1.18.x.mdx:26]: Not sure if this matters, but just to be safe:
```suggestion
endpoint, will result in the following warning from Vault:

<CodeBlockConfig hideClipboard>
```

File: website/content/docs/upgrading/upgrade-to-1.18.x.mdx
Comment (mpalmi) [inline website/content/docs/upgrading/upgrade-to-1.18.x.mdx:28]: It might be more helpful to show an actual request/output, as a user might see it.

Not really a blocker, so if it's a pain I can approve once the other items are addressed.

File: changelog/27350.txt
Comment (mpalmi) [inline changelog/27350.txt:3]:
```suggestion
when querying the activity log endpoints.
```

**Review by mpalmi — CHANGES_REQUESTED** [review]

**Review by mpalmi — APPROVED** [review]
Good stuff, @divyaac! Looks like this is ready to merge.

**Review by divyaac — COMMENTED** [review]
Addressed the changes with the resolved commits!

**Review by schavis — APPROVED** [review]

**Pull request merged**
